tsl/src/chunk_copy.c

```diff
@@ -442,29 +442,52 @@ chunk_copy_stage_sync_start(ChunkCopy *cc)
 
 	/* Start data transfer on the destination node */
 	cmd = psprintf("ALTER SUBSCRIPTION %s ENABLE", NameStr(cc->fd.operation_id));
-	ts_dist_cmd_run_on_data_nodes(cmd, list_make1(NameStr(cc->fd.dest_node_name)), false);
+	ts_dist_cmd_run_on_data_nodes(cmd, list_make1(NameStr(cc->fd.dest_node_name)), true);
 }
 
 static void
 chunk_copy_stage_sync(ChunkCopy *cc)
 {
-	const char *cmd;
+	char *cmd;
+
+	/*
+	 * Transaction blocks run in REPEATABLE READ mode in the connection pool.
+	 * However this wait_subscription_sync procedure needs to refresh the subcription
+	 * sync status data and hence needs a READ COMMITTED transaction isolation
+	 * level for that.
+	 */
+	cmd = psprintf("SET transaction_isolation TO 'READ COMMITTED'");
+	ts_dist_cmd_run_on_data_nodes(cmd, list_make1(NameStr(cc->fd.dest_node_name)), true);
+	pfree(cmd);
 
 	/* Wait until data transfer finishes in its own transaction */
 	cmd = psprintf("CALL _timescaledb_internal.wait_subscription_sync(%s, %s)",
 				   quote_literal_cstr(NameStr(cc->chunk->fd.schema_name)),
 				   quote_literal_cstr(NameStr(cc->chunk->fd.table_name)));
 
-	ts_dist_cmd_run_on_data_nodes(cmd, list_make1(NameStr(cc->fd.dest_node_name)), false);
+	ts_dist_cmd_run_on_data_nodes(cmd, list_make1(NameStr(cc->fd.dest_node_name)), true);
+	pfree(cmd);
 }
 
 static void
 chunk_copy_stage_drop_subscription(ChunkCopy *cc)
 {
-	const char *cmd;
+	char *cmd;
 
+	/* Stop data transfer on the destination node */
+	cmd = psprintf("ALTER SUBSCRIPTION %s DISABLE", NameStr(cc->fd.operation_id));
```

Comment:
It might makes sense to separate all of those steps in three separate stages, since we have a separate stage to create the replication slot

Reply:
We created these additional substeps just to allow things to work in a transaction block. Semantically it's all about dropping the subscription.

Reply:
Yes, but from point of view of the healing function which way is more convenient to operate?

Reply:
it's not an issue. The cleanup function will be exactly the same as this one with these 3 steps part of a transaction block